wal/wal_test.go

```diff
@@ -370,56 +370,123 @@ func TestSegmentMetric(t *testing.T) {
 	testutil.Ok(t, w.Close())
 }
 
-func BenchmarkWAL_LogBatched(b *testing.B) {
-	dir, err := ioutil.TempDir("", "bench_logbatch")
-	testutil.Ok(b, err)
+func TestCompression(t *testing.T) {
```

Comment:
Just a personal preference , but when the bootstrap is similar I usually do it like:

```
func TestCompression(t *testing.T) {
	boostrap := func(compressed bool) string {
		const (
			segmentSize = pageSize
			recordSize  = (pageSize / 2) - recordHeaderSize
			records     = 100
		)

		dirPath, err := ioutil.TempDir("", "")
		testutil.Ok(t, err)

		w, err := NewSize(nil, nil, dirPath, segmentSize, compressed)
		testutil.Ok(t, err)

		buf := make([]byte, recordSize)
		for i := 0; i < records; i++ {
			testutil.Ok(t, w.Log(buf))
		}
		testutil.Ok(t, w.Close())

		return dirPath
	}

	dirCompressed := boostrap(true)
	defer func() {
		testutil.Ok(t, os.RemoveAll(dirCompressed))
	}()
	dirUnCompressed := boostrap(false)
	defer func() {
		testutil.Ok(t, os.RemoveAll(dirUnCompressed))
	}()

	uncompressedSize, err := dirSize(dirUnCompressed)
	testutil.Ok(t, err)
	compressedSize, err := dirSize(dirCompressed)
	testutil.Ok(t, err)

	testutil.Assert(t, float64(uncompressedSize)*0.75 == float64(compressedSize), "Compressing zeroes should save at least 25%% space - uncompressedSize:%v, compressedSize:%v ", uncompressedSize, compressedSize)
}
```
this also removes the need for the 'logZeroRecords' helper

you can keep as per your suggestion if you prefer.

Reply:
I like your way more and got rid of the `logZeroRecords` helper. Thanks for the suggestion!